File: Sen/Implement Queue by Two Stacks.py
class Queue:

    def __init__(self):
        self.stack1 = []
        self.stack2 = []

    def adjust(self):
        if len(self.stack2) == 0:
            while len(self.stack1) != 0:
                self.stack2.append(self.stack1.pop())

    def push(self, element):
        # write your code here
        self.stack1.append(element)

    def top(self):
        # write your code here
        # return the top element
        self.adjust()
        return self.stack2[len(self.stack2) - 1]

    def pop(self):
        # write your code here
        # pop and return the top element
        self.adjust()
        return self.stack2.pop()

File: ming/0922/Implement Queue by Two Stacks.java
public class Solution {
    private Stack<Integer> stack1;
    private Stack<Integer> stack2;

    public Solution() {
       stack1 = new Stack();
       stack2 = new Stack();
    }
    
    public void push(int element) {
        stack1.push(element);
    }

    public int pop() {
        if (stack2.isEmpty()){
            while(!stack1.isEmpty()){
                stack2.push(stack1.pop()); 
            }
        }
        return stack2.pop();
    }

    public int top() {
        if (stack2.isEmpty()){
            while(!stack1.isEmpty()){
                stack2.push(stack1.pop()); 
            }
        }
        return stack2.peek();
    }
}

